### PR TITLE
 Fixes #5 Optimized the get_repositories functionality

### LINKED ISSUE #5: code: handle pagination when saving user repositories
Github API uses pagination and lists only 30 items per page by default. But a user can have more than 30 repositories. Handle this situation in the `get_repositories` function defined in the [utils.py](https://github.com/ankan17/contrib/blob/master/core/utils.py) file. Use the `per_page` and the `page` options for traversing.

### PR DESCRIPTION
**Pull request by JayjeetAtGithub**

### CODE REVIEW

File: core/utils.py [hunk 1]
```diff
@@ -7,16 +7,29 @@
...
 
 
+"""Generates the request URL based on the user details"""
+def get_request_url(username):
+
+	user_details_url = 'https://api.github.com/users/{}'.format(username)
+	response = requests.get(user_details_url)
+	if response.status_code == 200:
+		public_repos = response.json()['public_repos']
+		repos_url = 'https://api.github.com/users/{}/repos?per_page={}'.format(username,public_repos)
```
Comment (ankan17) [inline core/utils.py]: It's a better approach but still it has two flaws:
1. You are making at least two network calls even if the number of repositories is very less.
2. Even if you do `per_page=1000` GitHub will only send 100. That is the maximum. So, if a user has more than 100 repositories, this code will fail to grab it.

**Review by ankan17 — CHANGES_REQUESTED** [review]
It is an improvement upon your previous approach, but it can still be optimised. Also look at the travis logs to see why the tests are failing and fix those issues.

**Comment by ankan17** [comment]
Also do not close pull requests and open another for making further changes, rather push commits to your fork to update the PR.

File: core/utils.py [hunk 2]
```diff
@@ -7,29 +7,23 @@
...
-		repos_url = 'https://api.github.com/users/{}/repos?per_page={}'.format(username,public_repos)
-		return repos_url
-
-
-
 def get_repositories(username):
     """Returns the repositories of a user using the Github API."""
-    repos_url = get_request_url(username)
-    response = requests.get(repos_url)
+    repos_url = 'https://api.github.com/users/{}/repos?per_page=100&page='.format(username)
```
Comment (ankan17) [inline core/utils.py]: Change this to: `repos_url = 'https://api.github.com/users/{}/repos?per_page=100&page={}'`

File: core/utils.py [hunk 3]
```diff
@@ -7,29 +7,23 @@
...
+    page = 1
     repos = []
-    if response.status_code == 200:
-        repos = response.json()
+    while True:
+    	response = requests.get(repos_url+str(page))
```
Comment (ankan17) [inline core/utils.py]: Change this to: `requests.get(repos_url.format(username, page))`

File: core/utils.py [hunk 4]
```diff
@@ -7,29 +7,23 @@
...
+    	if response.status_code == 200:
+    		if len(response.json()) == 0:
```
Comment (ankan17) [inline core/utils.py]: This section can be simplified as:
```python
repos.extend(response.json())
if len(response.json()) < 100:
    break
```

**Review by ankan17 — CHANGES_REQUESTED** [review]
Looking good. A few changes and it will be good to go. Also please fix the tests while you are at it.

**Comment by JayjeetAtGithub** [comment]
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated

**Comment by JayjeetAtGithub** [comment]
Thanks a lot...

**Pull request merged**